### PR TITLE
Prevent excluded descendants from being picked as children

### PR DESCRIPTION
**Pull request by papandreou**

### CODE REVIEW

**Comment by sunesimonsen** [comment]
Mhh I though I did that here https://github.com/sunesimonsen/html-generators/blob/master/src/ElementGenerator.js#L149, but maybe that isn't enough. Didn't really do a lot of tests for it yet.

File: src/ElementGenerator.js [hunk 1]
```diff
@@ -112,17 +111,19 @@ class ElementGenerator extends Generator {
...
-
-      return {
-        value,
-        category: item.startsWith("@"),
-        required: requiredContent.has(value),
-        allowMultiple
-      };
-    });
+    )
+      .filter(item => !excludedDescendants.has(item))
```
Comment (sunesimonsen) [inline src/ElementGenerator.js]: The item might end with a `?`.

**Comment by sunesimonsen** [comment]
What happens if we exclude required children 🤔

**Comment by sunesimonsen** [comment]
I think this solution would work with the regex as it filters down the permitted content and my code filters the items that is generated from categories. Alternatively we could do this https://github.com/sunesimonsen/html-generators/compare/ssimonsen/exclude-whitelisted-children?expand=1

**Comment by papandreou** [comment]
Let's go with your fix instead! I just noticed that I was still getting `<script>`s inside `<picture>` and wanted to fix that :)

**Comment by papandreou** [comment]
As for required children, I think excluded descendants should take precedence, even if that means that we're missing required children. That also happens to be exactly how I need it to behave right now, so I might be biased :innocent:

**Comment by papandreou** [comment]
Cool! I added your fix for the `?` thing along with a test. Merging :rocket:

**Pull request merged**